featurebyte/feast/model/online_store.py

```diff
@@ -89,4 +102,12 @@ def get_feast_online_store_details(
     -------
     FeastOnlineStoreDetails
     """
-    return parse_obj_as(FeastOnlineStoreDetails, online_store_details.dict())  # type: ignore
+    params = online_store_details.dict(by_alias=True)
+    online_store_detail_class = ONLINE_STORE_DETAILS_CLASS_MAP.get(params.get("type"))
+    if online_store_detail_class is None:
+        # use pydantic builtin version to throw validation error (slow due to pydantic V2 performance issue)
+        return TypeAdapter(FeastOnlineStoreDetails).validate_python(params)
+
+    # use internal method to avoid current pydantic V2 performance issue due to _core_utils.py:walk
+    # https://github.com/pydantic/pydantic/issues/6768
+    return cast(FeastOnlineStoreDetails, online_store_detail_class(**params))
```

Comment:
This logic of using class map to construct an object seems to be repeated in a few places. Would it make sense to extract a helper function for reuse?